### PR TITLE
Label Set and User Group gridview delete confirmation

### PR DESCRIPTION
**Pull request by mfavetti**
@LouisGac 

I adopted the javascript approach that the other admin pages used to confirm deletion. This does not use bootstrap modals. I didn't want to make these 2 pages different from all the others. However, if you think modals would be better, I can work on changing all of the admin pages over to using modals for confirmations rather than the browser dialog.

### CODE REVIEW

**Comment by olleharstedt** [comment]
Personally, I would consider this Bootstrap solution: https://ethaizone.github.io/Bootstrap-Confirmation/

Might be easier than using modals.

File: application/views/admin/labels/labelsets_view.php [hunk 1]
```diff
@@ -69,11 +69,19 @@
...
-    $.fn.yiiGridView.update('labelsets-grid',{ data:{ pageSize: $(this).val() }});
+    // To update rows per page via ajax
+    $(document).on("change", '#pageSize', function() {
+        $.fn.yiiGridView.update('labelsets-grid',{ data:{ pageSize: $(this).val() }});
+    });
+    //Delete button
+    $(document).ready(function() {
+        $('a[data-confirm]').click(function() {
+            if (!confirm($(this).attr('data-confirm'))) {
+              return false;
```
Comment (olleharstedt) [inline application/views/admin/labels/labelsets_view.php]: `return false/true` is usually not necessary. Why not `return !confirm(...)`?

**Comment by SamMousa** [comment]
Just use a data attribute and bind some javascript to it to use whatever modal you want. Regardless I don't see the upside of reworking all admin pages now and then having to redo it for 3.0...

**Comment by Shnoulle** [comment]
> I adopted the javascript approach that the other admin pages used to confirm deletion
Yes : think it's the good way
> Just use a data attribute and bind some javascript
Yes, and have a js file , maybe in extension to do this . One function : samle function for whole parts and only one file to update for 3.0

https://github.com/LimeSurvey/LimeSurvey/pull/438/files#diff-43abaeb2f5e28c47bc7d3b48fd8e6540R78
This is not in a external file ?

**Comment by LouisGac** [comment]
Hi Matthew, thank you again for your great work, it's highly appreciated.

An homogenization of all the alert/modals/dialog/message-box in LimeSurvey is for sure absolutely needed. 

Right now, we have : 
1. **javascript alerts** : they are easy to replace
2. **jquery dialogs** : they can be tricky to remove when linked to jQgrid. They should be replaced by Bootstrap modals (see what have been done for subquestions/answers quick add)
3. **Bootstrap alerts** : they are used for Yii flash messages, they should be kept.
4. **Bootstrap modals** : should be generalized
5. **LimeSurvey message box** using bootstrap jumbotron (like in survey activation). Some of them must be kept, some of them must be replaced by modals, some of them must be replaced by alerts.

The case of the js confirmation alert is easy : they should be replaced by bootstrap modal. You can have a look in the home page settings, I used Bootstrap modal with a snippet of javascript. See : 
https://github.com/LimeSurvey/LimeSurvey/blob/master/application/models/Boxes.php#l140
https://github.com/LimeSurvey/LimeSurvey/blob/master/scripts/admin/homepagesettings.js#l69

But before generalizing this approach, it needs a second cycle of dev : 
1. The modal should be entirely translated ("Ok" and "Cancel"), and the text should be modular (Yes/No, confirm/cancel, continue/.. or whatever )
2. A view for the modal should be created, rather than using javascript HTML injection.
3. Because this view should be included in many different views, and because of point 1, a widget would be the perfect approach.

You can have a look about how is working the PannelBoxWidget, which is pretty simple  : 
https://github.com/LimeSurvey/LimeSurvey/tree/master/application/extensions/PannelBoxWidget

**Comment by Shnoulle** [comment]
@LouisGac : why not use a Yii extension ? Then any dev can use this extension to use the good method ?
And we can update more easily this extension.

**Comment by LouisGac** [comment]
well, we could create widgets for each case (with views of course ;-) )

but the main problem will remain : we'll have to find in each place of the code what case must be used. 

I think the most important is to replace message box with bootstrap alert messages when possible, and to remove all the JavaScript alerts.

**Comment by Shnoulle** [comment]
alert function can be easily globally replaced ;).

But before update to another function : have the widget must be done ? Else we do the task again when widget/extension is ready.

Denis

**Comment by LouisGac** [comment]
alert function can be easily globally replaced, as I explained above, I did it here for one specific case: 
https://github.com/LimeSurvey/LimeSurvey/blob/master/scripts/admin/homepagesettings.js#l69

But once you do it for all js alert, still needed to test all of them. 

About widgets or extensions or components, the important thing is that they must make the code easier to update, not harder. If we want to change the design again in the future (eg: upgrading to Bootstrap 4), it should be easy to update the HTML code. So views must be used for the rendering, and models to get the datas.

**Comment by Shnoulle** [comment]
OK, then adding a data-confirm with some html inside do a modal box. Great :)

A data + js solution is great :) .

**Comment by LouisGac** [comment]
if you read the discussion above, I say :
_'But before generalizing this approach, it needs a second cycle of dev : _' 
blablabla

**Comment by olleharstedt** [comment]
This commit puts a modal in the footer that can be used from everywhere in the admin: https://github.com/LimeSurvey/LimeSurvey/commit/e3596fffcf4f075a263243345507254660f043e5

**Pull request merged**